static/src/javascripts/projects/common/modules/discussion/comment-box.js

```diff
@@ -0,0 +1,593 @@
+                <i class="i i-alert"></i>
+                <span class="d-discussion__error-text">
+                    ${message || this.errorMessages[type]}
+                </span>
+            </div>`;
+        this.getElem('messages').appendChild(bonzo.create(errorMessage)[0]);
+        this.errors.push(type);
+    }
+
+    postComment(): void {
+        const comment: commentType = {
+            body: this.elem.body.value,
+        };
+
+        this.clearErrors();
+
+        const postCommentToDAPI = () => {
+            this.removeState('onboarding-visible');
+            comment.body = urlify(comment.body);
+            this.setFormState(true);
+            DiscussionApi.postComment(this.getDiscussionId(), comment)
+                .then((resp: Object) => this.postCommentSuccess(comment, resp))
+                .catch((err: Object) => this.fail(err));
+        };
+
+        const updateUsernameSuccess = (resp: Object): void => {
+            mediator.emit(
+                'user:username:updated',
+                resp.user.publicFields.username
+            );
+            this.options.hasUsername = true;
+            this.getElem('onboarding-username').classList.add('is-hidden');
+            postCommentToDAPI();
+        };
+
+        const updateUsernameFailure = (errorResponse: Object): void => {
+            const usernameField = this.getElem('onboarding-username-input');
+            const errorMessage = this.getElem(
+                'onboarding-username-error-message'
+            );
+
+            this.setState('onboarding-visible');
+
+            usernameField.classList.add(
+                'd-comment-box__onboarding-username-error-border'
+            );
+            errorMessage.innerHTML = JSON.parse(
+                errorResponse.responseText
+            ).errors[0].description;
+            errorMessage.classList.remove('is-hidden');
+        };
+
+        const validEmailCommentSubmission = () => {
+            if (comment.body === '') {
+                this.error('EMPTY_COMMENT_BODY');
+            }
+
+            if (comment.body.length > this.options.maxLength) {
+                this.error(
+                    'COMMENT_TOO_LONG',
+                    `<b>Comments must be shorter than ${this.options
+                        .maxLength} characters.</b>` +
+                        `Yours is currently ${comment.body.length -
+                            this.options.maxLength} character(s) too long.`
+                );
+            }
+
+            if (this.options.replyTo) {
+                comment.replyTo = this.options.replyTo;
+            }
+
+            if (this.errors.length === 0) {
+                if (this.options.newCommenter && !this.options.hasUsername) {
+                    IdentityApi.updateUsername(
+                        this.getElem('onboarding-username-input').value
+                    ).then(updateUsernameSuccess, updateUsernameFailure);
```

Comment:
There should arguably be another `catch` here in case the `JSON.parse` in `updateUsernameFailure` fails. Although not strictly part of this conversion, we can add a TODO to add it later.